src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -281,6 +282,18 @@ public ResponseEntity<JobModel> deleteFile(
     return jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
 
+  @Override
+  public ResponseEntity<List<FileModel>> listFiles(
+      @PathVariable("id") UUID id,
+      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit) {
```

Comment:
Copying down annotations from the base class doesn't have a benefit and it can lead to problems, so I would avoid doing it. There's no benefit because the annotations in the base class will be used if the subclass doesn't have any. It can lead to problems because there's no warning if the base class annotations change and conflict with the subclass annotations. In either case you want the base class annotations to be used, since they're the ones generated by the openapi code generator.

```suggestion
  public ResponseEntity<List<FileModel>> listFiles(UUID id, Integer offset, Integer limit) {
```